scripts/check-update.sh

```diff
@@ -26,15 +54,42 @@ cp release-$ami_type.auto.pkrvars.hcl release-$ami_type.old.hcl
 set +e
 diff_val=$(diff <(grep -v ami_version release-$ami_type.old.hcl) <(grep -v ami_version release-$ami_type.auto.pkrvars.hcl))
 set -e
+
+# Check for AL2023 NVIDIA driver version
+if [ "$ami_type" = "al2023" ]; then
+    gpu_update=$(./scripts/check-update-security.sh "${ami_type}_gpu")
+    handle_nvidia_version "$ami_type" "$gpu_update"
+fi
+
 # If no difference in dependencies, check for security update
 if [ -z "$diff_val" ]; then
-    # al2023 version already generates a diff in dependency file if it has security updates, so no check necessary if al2023
-    if [ "$ami_type" != "al2023" ]; then
+    Update="false"
+    case "$ami_type" in
+    "al2023")
+        # AL2023 version already generates a diff in dependency file if it has security updates, so no check necessary if AL2023
```

Comment:
do we need to call `check-update-security.sh` for al2023 gpu? Because that script seems to have been updated but is not being called with al2023_gpu parameter